src/components/ProductItem/ProductItem.tsx

```diff
@@ -0,0 +1,128 @@
+import React from "react";
+import { useState, useEffect } from "react";
+
+import Button from "@components/Button";
+import { ButtonColor } from "@components/Button";
+import Card from "@components/Card";
+import axios from "axios";
+import { Link, useParams } from "react-router-dom";
+
+import styles from "./ProductItem.module.scss";
+
+interface IProduct {
+  id: number;
+  image: string;
+  title: string;
+  subtitle: string;
+  category: string;
+  content: string;
+}
+const ProductItem = () => {
+  const [allproducts, setAllProducts] = useState([]);
+  useEffect(() => {
+    const fetchAllproducts = async () => {
+      const result = await axios({
+        method: "get",
+        url: "https://fakestoreapi.com/products",
+      });
+      setAllProducts(
+        result.data.map(
+          (raw: {
+            id: number;
+            image: string;
+            title: string;
+            description: string;
+            category: string;
+            price: number;
+          }) => ({
+            id: raw.id,
+            image: raw.image,
+            title: raw.title,
+            subtitle: raw.description,
+            category: raw.category,
+            content: raw.price,
+          })
+        )
+      );
+    };
+    fetchAllproducts();
+  }, []);
+
+  const { id } = useParams();
+  const [product, setProduct] = useState<IProduct>();
+  useEffect(() => {
+    const fetch = async () => {
+      const result = await axios({
+        method: "get",
+        url: `https://fakestoreapi.com/products/${id}`,
+      });
+      const raw = result.data;
+      setProduct({
+        id: raw.id,
+        image: raw.image,
+        title: raw.title,
+        subtitle: raw.description,
+        category: raw.category,
+        content: raw.price,
+      });
+    };
+    fetch();
+  }, [id]);
+  let relatedProducts = allproducts.filter(
+    (item: IProduct) => item.category === product?.category
+  );
+  relatedProducts.splice(3);
```

Comment:
тут стоит использовать [хук useMemo](https://reactjs.org/docs/hooks-reference.html#usememo). и почему ты тут сразу не вызвал метод `splice` на `allProducts`?
```suggestion
  const relatedProducts = allproducts
    .filter((item: IProduct) => item.category === product?.category )
    .splice(3);
```